pretyping/unification.ml

```diff
@@ -2377,15 +2377,23 @@ type head_kind =
 | HeadOther
 
 let get_head_kind ~metas env sigma c =
-  match get_type_of_with_metas ~lax:true ~metas env sigma c with
-  | ty ->
+  let ty = get_type_of_with_metas ~lax:true ~metas env sigma c in
```

Comment:
why lax:true when we stop catching retype errors?

Reply:
We didn't catch the errors, due to the crappy pattern-match syntax of OCaml the branch was actually associated to the inner match and thus dead code.